### PR TITLE
Fixed contact update after propertyGroup refactor

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ <skjnldsv@protonmail.com>

### CODE REVIEW

File: templates/contactDetails.html [hunk 1]
```diff
@@ -42,7 +42,7 @@
 			</div>
 		</header>
 		<section>
-			<propertyGroup ng-repeat="prop in ctrl.contact.props | toArray | orderDetailItems:'$key'" model="ctrl.contact" data="prop" name="prop.$key"></propertyGroup>
+			<propertyGroup ng-repeat="prop in ctrl.contact.props | toArray | orderDetailItems:'$key'" model="ctrl.contact" addressBook="ctrl.addressBook" data="prop" name="prop.$key"></propertyGroup>
```
Comment (skjnldsv) [inline templates/contactDetails.html]: @Henni this doesn't work and fails with a `Error: [$compile:multidir]`
Can you help?

Comment (Henni) [reply templates/contactDetails.html]: Adding the attribute `addressBook` breaks this.
We have a directive which is called `addressBook`, so Angular doesn't know if it should now render the addressBook or the propertyGroup directive.
You have to choose a different attribute name.

File: js/components/detailsItem/detailsItem_directive.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ angular.module('contactsApp')
 			name: '=',
 			data: '=',
 			contact: '=model',
+			addressBook: '=addressBook',
```
Comment (Henni) [inline js/components/detailsItem/detailsItem_directive.js]: Just FYI this line is to specific: `addressBook: '='` would suffice.
You only need to write something after the `=` if the attribute should be named differently to the controller variable.

File: templates/detailItems/adr.html [hunk 3]
```diff
@@ -1,36 +1,36 @@
-<select ng-model="ctrl.type" ng-model-options="{ debounce: 500 }" ng-change="ctrl.changeType(ctrl.type)" ng-disabled="ctrl.model.addressBook.readOnly">
+<select ng-model="ctrl.type" ng-model-options="{ debounce: 500 }" ng-change="ctrl.changeType(ctrl.type)" ng-disabled="ctrl.addressBook.readOnly">
 	<option ng-repeat="option in ctrl.availableOptions" value="{{option.id}}">{{option.name}}</option>
 </select>
-<button ng-click="ctrl.deleteField()" class="icon-delete" title="{{ctrl.t.delete}}" ng-if="!ctrl.model.addressBook.readOnly"></button>
+<button ng-click="ctrl.deleteField()" class="icon-delete" title="{{ctrl.t.delete}}" ng-if="!ctrl.addressBook.readOnly"></button>
```
Comment (Henni) [inline templates/detailItems/adr.html]: I would suggest that we offer a readOnly property for contacts, which reflects the readOnly property of the addressbook.
This way we also don't have to pass the addressbook to the propertyGroup

**Comment by skjnldsv** [comment]
All clear! @irgendwie @Henni Please review! :)
cc @MorrisJobke @jancborchardt

**Comment by MorrisJobke** [comment]
> Code changes seem reasonable, every field update gets pushed to the server and read-only view is also working again! 👍

Then merge it ;)

**Comment by irgendwie** [comment]
@MorrisJobke Was waiting for another :+1: 😉

**Pull request merged**